### PR TITLE
feat(hooks): expose a ssh public key hook for builder to interact with

### LINKED ISSUE #351: add new url to interact with deis/builder
*Request:*

`GET /v2/hooks/public_key/{ssh_public_key}`

(where ssh_public_key is the SSH public key as received by the builder in Base64)

*Response:*
```
{
  "username": "john.doe",
  "fingerprint": "",
  "apps": [
    "demo"
  ]
}
```

*Where:*
- `fingerprint` is the stored value in `api_key`
- `repositories` an array that contains all the apps where the user can run `git push`

*Return codes:*
- 200 if the key is valid with the response
- 404 if the key is not a valid one

### PR DESCRIPTION
**Pull request by helgi**
Fixes #351

### CODE REVIEW

**Comment by aledbf** [comment]
Code LGTM

File: rootfs/api/tests/test_hooks.py [hunk 1]
```diff
@@ -105,6 +106,28 @@ def test_key_hook(self):
             {'key': public2, 'fingerprint': '43:fd:22:bc:dc:ca:6a:28:ba:71:4c:18:41:1d:d1:e2'}
         ]})
 
+        # Fetch a valid ssh key
+        url = '/v2/hooks/key/{}'.format(base64.urlsafe_b64encode(public.encode('utf-8')).decode())
```
Comment (slack) [inline rootfs/api/tests/test_hooks.py]: Any reason not to use the fingerprint of the key rather than the entire key?

Comment (helgi) [inline rootfs/api/tests/test_hooks.py]: @aledbf would have to answer that. He outlined his spec in #351

**Comment by aledbf** [comment]
@slack the only "reason" is that only the pub key is available. I can add a fingerprint function and pass that instead of the key

File: rootfs/api/tests/test_hooks.py (hunk 1 above)
Comment (arschles) [inline rootfs/api/tests/test_hooks.py]: I believe that using the encoded key is good enough for now, but we should, in the future, switch to a fingerprint. Doing so will at least further obfuscate identifying information in the public key. https://github.com/deis/builder/issues/158 tracks

**Comment by arschles** [comment]
@helgi  I tested this with a build of https://github.com/aledbf/builder/pull/1 and verified it works

**Pull request merged**